pkg/storage/snapshot/restore.go

```diff
@@ -139,6 +139,11 @@ func (ctrl *VMRestoreController) updateVMRestore(vmRestoreIn *snapshotv1.Virtual
 	}
 	controller.AddFinalizer(vmRestoreOut, vmRestoreFinalizer)
 
+	err = ctrl.updateVolumeRestores(vmRestoreOut)
```

Comment:
Not sure about moving this here. Feels a little awkward. The function up to this point is about initializing the vmrestore before doing the real work. Also in #12601 want to be more deliberate about when when metadata/spec and status are updated and this is muddying things up even more